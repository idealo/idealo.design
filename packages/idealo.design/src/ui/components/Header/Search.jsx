import React from "react";
import s from "./Header.module.scss";
import CloseIco from "./ico_cross_circle_outline.svg";
import MagnifierIco from "./ico_search.svg";
import { fetchUserInfo } from "../../pages/BlogPage/data";
import withStyles from "isomorphic-style-loader/withStyles";

class Search extends React.Component {

    constructor(props) {
        super(props)

        this.handleOnKeyUp = this.handleOnKeyUp.bind(this)

        this.state = {
            isLoggedIn: false,
            userInfo: {},
            initialString: ''
        }

    }


    async componentDidMount() {
        window.document.addEventListener('keyup', this.handleOnKeyUp)
        try{
            const userInfo = await fetchUserInfo()
            this.setState({userInfo:userInfo})
            const isLoggedIn = userInfo.status === 'LOGGED_IN'
            this.setState({isLoggedIn:isLoggedIn})
            if(this.state.isLoggedIn){
                const displayName = userInfo.user['displayName'].charAt(0).toUpperCase();
                const surname = userInfo.user['surname'].charAt(0).toUpperCase();
                const initialString = displayName + surname;
                this.setState({initialString})
            }
        }catch (error){
        }
    }

    componentWillUnmount() {
        window.document.removeEventListener('keyup', this.handleOnKeyUp)
    }

    handleOnKeyUp(event) {
        if (event.ctrlKey && event.which == 70) {
            this.props.onClick()
        }
    }

    render() {
        const searchInputStyle = {
            visibility: this.props.isOpen ? 'visible' : 'hidden',
            width: this.props.isOpen ? '40vw' : 0,
            padding: this.props.isOpen ? '.5rem' : 0,
            margin: this.props.isOpen ? 'auto 2rem auto auto' : 0,
        }

        return (
            <>
                <input
                    style={searchInputStyle}
                    className={s.SearchInput}
                    onTransitionEnd={event => {
                        event.persist()
                        event.target.focus()
                        event.target.value = ''
                    }}
                    autoFocus/>

                {this.props.isOpen ?
                    <CloseIco className={s.SearchToggle} onClick={this.props.onClick}/> :
                    <MagnifierIco className={s.SearchToggle} onClick={this.props.onClick}/>}

                {this.state.isLoggedIn ?
                    <div>
                        <button className={s.initialsStyle}>{this.state.initialString}</button>
                        <a href="/logout"><button className={s.logoutButton} title="logoutButton"><span>Logout</span></button></a>
                    </div> :
                    <a href="/auth/provider"><button className={s.loginButton} title="loginButton"><span>Login</span></button></a>
                }

            </>
        )
    }
}

export default Search;
export { s as styles };