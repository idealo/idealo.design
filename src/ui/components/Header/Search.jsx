import React from "react";
import s from "./Header.module.scss";
import CloseIco from "./ico_cross_circle_outline.svg";
import MagnifierIco from "./ico_search.svg";
import {fetchUserInfo} from "../../pages/BlogPage/data";

class Search extends React.Component {

    constructor(props) {
        super(props)

        this.handleOnKeyUp = this.handleOnKeyUp.bind(this)
        this.handleLoginLogout = this.handleLoginLogout.bind(this)

        this.state = {
            isLoggedIn: false,
            userInfo: {},
            initialString: ''
        }

    }


    async componentDidMount() {
        window.document.addEventListener('keyup', this.handleOnKeyUp)
        try {
            const userInfo = await fetchUserInfo()
            this.setState({userInfo: userInfo})
            const isLoggedIn = userInfo.status === 'LOGGED_IN'
            this.setState({isLoggedIn: isLoggedIn})
            if (this.state.isLoggedIn) {
                const displayName = userInfo.user['displayName'].charAt(0).toUpperCase();
                const surname = userInfo.user['surname'].charAt(0).toUpperCase();
                const initialString = displayName + surname;
                this.setState({initialString})
            }
        } catch (error) {
        }
    }

    componentWillUnmount() {
        window.document.removeEventListener('keyup', this.handleOnKeyUp)
    }

    handleOnKeyUp(event) {
        if (event.ctrlKey && event.which == 70) {
            this.props.onClick()
        }
    }

    handleLoginLogout(){
        if (typeof window !== 'undefined') {
            localStorage.setItem('lastVisitedPage', window.location.pathname)
        }
        /*if (typeof window !== 'undefined') {
            const name = "lastVisitedPage"
            const value = window.location.pathname
            document.cookie = name + "=" + value + ";"
            console.log(document)
        }*/
    }

    render() {
        const searchInputStyle = {
            visibility: this.props.isOpen ? 'visible' : 'hidden',
            width: this.props.isOpen ? '40vw' : 0,
            padding: this.props.isOpen ? '.5rem' : 0,
            margin: this.props.isOpen ? 'auto 2rem auto auto' : 0,
        }

        return (
            <>
                <input
                    style={searchInputStyle}
                    className={s.SearchInput}
                    onTransitionEnd={event => {
                        event.persist()
                        event.target.focus()
                        event.target.value = ''
                    }}
                    autoFocus/>

                {this.props.isOpen ?
                    <CloseIco className={s.SearchToggle} onClick={this.props.onClick}/> :
                    <MagnifierIco className={s.SearchToggle} onClick={this.props.onClick}/>}

                {this.state.isLoggedIn ?
                    <div className={s.loginLogout} style={{display:'flex'}}>
                        <button style={{margin:'10px'}} className={s.initialsStyle}>{this.state.initialString}</button>
                        <a href="/logout">
                            <button onClick={this.handleLoginLogout} className={s.button} title="logoutButton"><span>Logout</span></button>
                        </a>
                    </div> :
                    <div className={s.loginLogout}>
                        <a href="/auth/provider">
                            <button onClick={this.handleLoginLogout} className={s.button} title="loginButton"><span>Login</span></button>
                        </a>
                    </div>
                }
            </>
        )
    }
}

export default Search;
export {s as styles};