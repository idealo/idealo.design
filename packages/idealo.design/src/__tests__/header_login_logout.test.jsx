import React from 'react';
import { create } from "react-test-renderer"
import Search from '../ui/components/Header/Header';

it('login renders correctly', () => {
    const logoutTest = create(<Search/>);
    expect(logoutTest.toJSON()).toMatchSnapshot();
});

it('logout renders correctly', () => {
    const logoutTest = create(<Search />);
    const instance = logoutTest.getInstance();
    expect(instance.state.isLoggedIn).toBe(false);
    instance.setState({isLoggedIn:true});
    expect(instance.state.isLoggedIn).toBe(true);
    expect(logoutTest.toJSON()).toMatchSnapshot();
});


/*
class Button extends React.Component {
    constructor(props) {
        super(props);
        this.state = { text: "" };
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
        this.setState(() => {
            return { text: "PROCEED TO CHECKOUT" };
        });
    }

    render() {
        return (
            <button onClick={this.handleClick}>
                {this.state.text || this.props.text}
            </button>
        );
    }
}

describe("Button component", () => {
    test("it shows the expected text when clicked (testing the wrong way!)", () => {
        const component = create(<Button text="SUBSCRIBE TO BASIC" />);
        const instance = component.getInstance();
        expect(instance.state.text).toBe("");
    });
});
*/
