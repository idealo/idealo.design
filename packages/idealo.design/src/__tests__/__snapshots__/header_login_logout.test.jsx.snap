// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders correctly 1`] = `
<header
  className="Header"
  style={null}
>
  <svg
    className="SideNavToggle"
    height="24"
    onClick={[Function]}
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      ico__hamburger
    </title>
    <path
      d="M22 13H2a1 1 0 0 1 0-2h20a1 1 0 0 1 0 2zm1 5a1 1 0 0 0-1-1H2a1 1 0 0 0 0 2h20a1 1 0 0 0 1-1zm0-12a1 1 0 0 0-1-1H2a1 1 0 0 0 0 2h20a1 1 0 0 0 1-1z"
      fill="#fff"
    />
  </svg>
  <h1
    style={null}
  >
    <a
      href="/"
      onClick={[Function]}
      style={null}
    >
      <span
        style={
          Object {
            "borderBottom": "1px solid orange",
          }
        }
      >
        idealo
      </span>
       
      <b>
        Design System
      </b>
    </a>
  </h1>
  <div
    className="StickyMenu"
    style={
      Object {
        "marginLeft": "17rem",
      }
    }
  />
  <input
    autoFocus={true}
    className="SearchInput"
    onTransitionEnd={[Function]}
    style={
      Object {
        "margin": 0,
        "padding": 0,
        "visibility": "hidden",
        "width": 0,
      }
    }
  />
  <svg
    className="SearchToggle"
    height="24"
    onClick={[Function]}
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <title>
      ico__search
    </title>
    <path
      d="M22.658 21.28L17.9 16.522a9.558 9.558 0 1 0-1.424 1.4l4.768 4.768a1 1 0 1 0 1.414-1.414zM3.049 10.513a7.5 7.5 0 1 1 7.5 7.5 7.509 7.509 0 0 1-7.5-7.5z"
      fill="#fff"
    />
  </svg>
  <a
    href="/auth/provider"
  >
    <button
      className="loginButton"
    >
      <span>
        Login
      </span>
    </button>
  </a>
</header>
`;
